### PR TITLE
feat:  TapFriends 3.7

### PR DESCRIPTION
**Pull request by weakish**

### CODE REVIEW

**Comment by 867380699** [comment]
您发给我的信件已经收到,谢谢!

**Review by onerain88 — APPROVED** [review]

File: docs/sdk/friends/follow.mdx [hunk 1]
```diff
@@ -401,25 +431,25 @@ await TDSFollows.FollowByShortCode(shortId);
...
 ```
 
 ```objc
-[TDSFollows followWithShortCode:shortId callback:^(BOOL succeeded, NSError * _Nullable error) {
-  if (succeeded) {
-    // Followed.
-  } else if (error) {
-    // Failed to follow.
+[TDSFollows followWithShortCode:shortId callback:^(TDSFriendHandleResult * _Nullable result, NSError * _Nullable error) {
+  if (error) {
```
Comment (zapcannon87) [inline docs/sdk/friends/follow.mdx:449]: 应该是这样：
```objc
if (error) {
  // handle error
} else {
  // handle result
}
```

File: docs/sdk/friends/follow.mdx [hunk 2]
```diff
@@ -765,28 +900,143 @@ TDSFollows.SetShareLink("https://please-replace-with-your-domain.example.com");
...
+```
+
+```objc
+NSError *error;
+TDSFriendLinkOption *option = [TDSFriendLinkOption new];
+option.roleName = @"她姥姥";
+option.queries = @{
+    @"ref" : @"taptap",
+};
+NSString *inviteUrl = [TDSFollows generateFriendInvitationLinkWithError:option error:&error];
```
Comment (zapcannon87) [inline docs/sdk/friends/follow.mdx:958]: `WithOption`

File: docs/sdk/friends/mutual.mdx [hunk 3]
```diff
@@ -996,36 +1075,74 @@ NSString *inviteUrl = [TDSFriends generateFriendInvitationLinkWithError:&error];
...
+```
+
+```objc
+NSError *error;
+TDSFriendLinkOption *option = [TDSFriendLinkOption new];
+option.roleName = @"她姥姥";
+option.queries = @{
+    @"ref" : @"taptap",
+};
+NSString *inviteUrl = [TDSFriends generateFriendInvitationLinkWithError:option error:&error];
```
Comment (zapcannon87) [inline docs/sdk/friends/mutual.mdx:1106]: `WithOption`

Comment (weakish) [reply docs/sdk/friends/mutual.mdx]: fixed in [50a8e40](https://github.com/taptap/tds-docs/pull/93/commits/50a8e40bcc73c87f4e0a114e863df41b6d109561)

File: docs/sdk/friends/follow.mdx (hunk 2 above)
Comment (weakish) [reply docs/sdk/friends/follow.mdx]: fixed in [50a8e40](https://github.com/taptap/tds-docs/pull/93/commits/50a8e40bcc73c87f4e0a114e863df41b6d109561)

File: docs/sdk/friends/follow.mdx (hunk 1 above)
Comment (weakish) [reply docs/sdk/friends/follow.mdx:449]: fixed in [50a8e40](https://github.com/taptap/tds-docs/pull/93/commits/50a8e40bcc73c87f4e0a114e863df41b6d109561)

**Review by zapcannon87 — APPROVED** [review]

**Review by fuchenshi — APPROVED** [review]

**Pull request merged**